docs/codeql/codeql-for-visual-studio-code/working-with-codeql-packs-in-visual-studio-code.rst

```diff
@@ -0,0 +1,30 @@
+:tocdepth: 1
+
+.. _working-with-codeql-packs-in-visual-studio-code:
+
+Working with CodeQL packs in Visual Studio Code
+===============================================
+
+.. include:: ../reusables/beta-note-package-management.rst
+
+You can view CodeQL packs and write and edit queries for them in Visual Studio Code. 
+
+About CodeQL packs
+------------------
+CodeQL packs are used to create, share, depend on, and run CodeQL queries and libraries. You can publish your own CodeQL packs and download packs created by others. For more information, see ":ref:`About CodeQL packs <about-codeql-packs>`".
+
+Creating and editing CodeQL packs in Visual Studio Code
+-------------------------------------------------------
+To create a new CodeQL pack, you will need to use the CodeQL CLI, which you can do within Visual Studio Code or outside of it with the ``codeql pack init`` command. Once you create an empty pack, you can edit the ``qlpack.yml`` file to add dependencies or change the name or version. For more information, see ":ref:`Creating and working with CodeQL packs <creating-and-working-with-codeql-packs>`".
+
+You can create or edit queries in a CodeQL pack in Visual Studio Code as you would with any CodeQL query, using the standard code editing features such as autocomplete suggestions to find elements to use from the pack's dependencies. 
```

Comment:
Do you think anyone would need a link here to more info? I don't know if it's necessary, but we could link to https://codeql.github.com/docs/codeql-for-visual-studio-code/setting-up-codeql-in-visual-studio-code/ for anyone who hasn't started using VS Code until they're reading this. Or possibly another link, but I think how this is written now lets people know what they can do in VS Code and is great while this is in beta 😄

Reply:
It's possible! I didn't think they would need to be linked to that, since they are already in that section of the docs and probably saw it, but its worth looking at a little more closely for sure, maybe it would be helpful!

I am also now wondering if I should link to this article from the CodeQL packs documentation on the CLI...

Reply:
> It's possible! I didn't think they would need to be linked to that, since they are already in that section of the docs and probably saw it

Good point and I agree.

> I am also now wondering if I should link to this article from the CodeQL packs documentation on the CLI...

I did a quick look and other areas of the docs that mention VS Code don't link to any VS Code docs. The VS Code sections are pretty visible in the sidebar, so I think this is good as is ✨

Reply:
great, thank you - I probably was overthinking it :) Sounds like we will leave out any other links for now!